optimum/habana/transformers/models/llama/modeling_llama.py

```diff
@@ -300,6 +300,13 @@ def get_k_proj_weight(self):
             return self.k_proj.qweight
         return self.k_proj.weight
 
+    def get_k_proj_weight_dtype(self):
+        """ 4bit quantization in GPTQ replaces the k_proj.weight with qweight.
+            Scales tensor gets the weight dtype. """
+        if hasattr(self.k_proj, 'qweight'):
+            return self.k_proj.scales.dtype
+        return self.k_proj.weight.dtype
+
```

Comment:
why do we need this duplicated function? can you call get_k_proj_weight().dtype?